src/ALCProxy.Communication/ICommObjects.cs

```diff
@@ -0,0 +1,257 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. 
+// The .NET Foundation licenses this file to you under the MIT license. 
+// See the LICENSE file in the project root for more information. 
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+
+namespace ALCProxy.Communication
+{
+    /// <summary>
+    /// This currently is designed to only work in-process
+    /// TODO: set up to allow for construction of out-of-proc proxies
+    /// </summary>
+    public abstract class ALCClient : IProxyClient
+    {
+        //Can't make this an IServerObject directly due to the type-loading barrier
+        protected object _server;
+        protected string _serverTypeName;
+        protected Type _intType;
+        protected MethodInfo _callMethod;
+        public ALCClient(Type interfaceType, string serverName)
+        {
+            _intType = interfaceType;
+            _serverTypeName = serverName;
+        }
+        private Type FindTypeInAssembly(string typeName, Assembly a)
+        {
+            //find the type we're looking for
+            Type t = null;
+            foreach (Type ty in a.GetTypes())
```

Comment:
Wondering if we can use `Assembly.GetType` instead of iterating through all types and matching ourselves.

Reply:
`Assembly.GetType` works, but only if we put the full name of the assembly (ex. entering "ALCProxy.Communication.ServerDispatch\`1" vs entering "ServerDispatch\`1") in the path, which the user may not do. I'll add ways for users to use both options and change this to use `Assembly.GetType` no matter which version of the type name you've used.